stablehlo/dialect/VhloOps.cpp

```diff
@@ -322,16 +322,16 @@ bool checkIfOperandAndResultElementTypesMatch(TypeRange operandTypes,
   if (llvm::any_of(
           llvm::zip(inputElementTypes, resultElementTypes),
           [&](auto pair) { return std::get<0>(pair) != std::get<1>(pair); }))
-    return true;
+    return false;
 
-  return false;
+  return true;
```

Comment:
Nit: I think this can just be `return !llvm::any_of(...);`

Reply:
Or at that point, might be clearer to do `llvm::all_of(... == ...)`

Reply:
done.